src/pypdfium2/_helpers/page.py

```diff
@@ -336,6 +340,131 @@ def flatten(self, flag=pdfium_c.FLAT_NORMALDISPLAY):
         return rc
     
     
+    # SPDX-SnippetBegin
+    # SPDX-License-Identifier: Apache-2.0 OR BSD-3-Clause
+    # SPDX-SnippetCopyrightText: 2025 Yizhan Huang <yizhanhuang2002@gmail.com>
+    def get_missing_fonts(self, textpage=None):
+        """
+        Check for non-embedded fonts on the page that may not render correctly.
+        
+        This method iterates through all text objects on the page and identifies fonts that are not embedded.
+        Since this involves multiple API calls, it may be expensive for pages with many text objects.
+        
+        Parameters:
+            textpage (PdfTextPage | None):
+                Optional textpage handle. If provided, it will be reused instead of creating a new one.
+                This can improve performance if you already have a textpage for other operations.
+        
+        Returns:
+            dict[str, list[str]]: Dictionary mapping font names (base names) to lists of text content samples that use them.
+                The dictionary will be empty if all fonts are embedded or if no text objects are found.
+        
+        Example:
+            >>> missing = page.get_missing_fonts()
+            >>> if missing:
+            ...     print(f"Found {len(missing)} non-embedded fonts")
+            
+            >>> # Reuse an existing textpage
+            >>> textpage = page.get_textpage()
+            >>> missing = page.get_missing_fonts(textpage=textpage)
+            >>> textpage.close()
+        """
+        missing_fonts = {}
+        textpage_created = False
+        try:
+            if textpage is None:
+                textpage = self.get_textpage()
+                textpage_created = True
+            
+            for obj in self.get_objects(filter=[pdfium_c.FPDF_PAGEOBJ_TEXT], textpage=textpage):
+                try:
+                    font = obj.get_font()
+                    if not font.get_is_embedded():
+                        font_name = font.get_base_name()
+                        
+                        text_content = None
+                        try:
+                            extracted = obj.extract()
+                            if extracted and extracted.strip():
+                                if len(extracted) > 100:
+                                    text_content = extracted[:97] + "..."
+                                else:
+                                    text_content = extracted
+                        except (PdfiumError, Exception):
+                            pass
+                        
+                        if font_name not in missing_fonts:
+                            missing_fonts[font_name] = []
+                        if text_content:
+                            missing_fonts[font_name].append(text_content)
+                except (PdfiumError, AttributeError):
+                    continue
+        except Exception:
+            pass
```

Comment:
I might prefer to avoid broad exception catching here.
In particular, I'd say `Exception` and `AttributeError` are too generic and could theoretically mask e.g. a typo in the code.
Also, I think there is some redundancy in your current try/except hierarchy.

Reply:
Even with `PdfiumError`, I'm not sure silently suppressing it is good.
Unless there's a concrete reason (like with `textpage.get_textobj()` and escape chars), it might fit the API design better not to do any exception catching here, and leave it up to the caller whether they want to guard for `PdfiumError`.